mmpose/models/detectors/mesh.py

```diff
@@ -0,0 +1,298 @@
+                return outputs
+
+        # optimize generator
+        pred_out = self.smpl(
+            betas=pred_beta,
+            body_pose=pred_pose[:, 1:],
+            global_orient=pred_pose[:, [0]],
+            pose2rot=False)
+        pred_vertices = pred_out.vertices
+        pred_joints_3d = self.get_3d_joints_from_mesh(pred_vertices)
+        gt_beta = data_batch['beta']
+        gt_pose = data_batch['pose']
+        gt_vertices = self.smpl(
+            betas=gt_beta,
+            body_pose=gt_pose[:, 3:],
+            global_orient=gt_pose[:, :3]).vertices
+        pred = dict(
+            pose=pred_pose,
+            beta=pred_beta,
+            camera=pred_camera,
+            vertices=pred_vertices,
+            joints_3d=pred_joints_3d)
+
+        target = {
+            key: data_batch[key]
+            for key in [
+                'pose', 'beta', 'has_smpl', 'joints_3d', 'joints_2d',
+                'joints_3d_visible', 'joints_2d_visible'
+            ]
+        }
+        target['vertices'] = gt_vertices
+
+        losses = self.loss_mesh(pred, target)
+
+        if self.with_gan:
+            set_requires_grad(self.discriminator, False)
+            pred_theta = (pred_camera, pred_pose, pred_beta)
+            pred_score = self.discriminator(pred_theta)
+            loss_adv = self.loss_gan(
+                pred_score, target_is_real=True, is_disc=False)
+            losses['adv_loss'] = loss_adv
+
+        loss, log_vars = self._parse_losses(losses)
+        optimizer['generator'].zero_grad()
+        loss.backward()
+        optimizer['generator'].step()
+
+        outputs = dict(
+            loss=loss,
+            log_vars=log_vars,
+            num_samples=len(next(iter(data_batch.values()))))
+
+        return outputs
+
+    def forward_train(self, *args, **kwargs):
+        """Forward function for training.
+
+        For ParametricMesh, we do not use this interface.
+        """
+        raise NotImplementedError('This interface should not be used in '
+                                  'current training schedule. Please use '
+                                  '`train_step` for training.')
+
+    def val_step(self, data_batch, **kwargs):
+        """Forward function for evaluation.
+
+        Args:
+            data_batch (dict): Contain data for forward.
+
+        Returns:
+            dict: Contain the results from model.
+        """
+        output = self.forward_test(**data_batch, **kwargs)
+        return output
+
+    def forward_test(self, img, img_metas, **kwargs):
+        """Defines the computation performed at every call when testing."""
+        assert img.size(0) == 1
+        assert len(img_metas) == 1
+
+        pred_smpl = self.generator(img)
+        pred_pose, pred_beta, pred_camera = pred_smpl
+        pred_out = self.smpl(
+            betas=pred_beta,
+            body_pose=pred_pose[:, 1:],
+            global_orient=pred_pose[:, [0]],
```

Comment:
`pred_pose[:, [0]]` will make a copy, and `pred_pose[:, 0:1]` is a slice without copy.

Reply:
OK, I will modify it to` pred_pose[:, 0].unsqueeze(1)`

Reply:
:1 is good enough